provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -961,6 +961,56 @@ func generateSchema() schema.PackageSpec {
 				InputProperties: vpcCniProperties(true /*kubeconfig*/),
 				RequiredInputs:  []string{"kubeconfig"},
 			},
+			"eks:index:Addon": {
+				IsComponent: true,
+				ObjectTypeSpec: schema.ObjectTypeSpec{
+					Description: "Addon manages an EKS add-on.\n" +
+						"For more information about supported add-ons, see: https://docs.aws.amazon.com/eks/latest/userguide/eks-add-ons.html",
+				},
+				InputProperties: map[string]schema.PropertySpec{
+					"addonName": {
+						TypeSpec:    schema.TypeSpec{Type: "string"},
+						Description: "Name of the EKS add-on. The name must match one of the names returned by describe-addon-versions.",
+					},
+					"addonVersion": {
+						TypeSpec:    schema.TypeSpec{Type: "string"},
+						Description: "The version of the EKS add-on. The version must match one of the versions returned by describe-addon-versions.",
+					},
+					"cluster": {
+						TypeSpec: schema.TypeSpec{
+							Ref: "#/resources/eks:index:Cluster",
+						},
+						Description: "The target EKS cluster.",
+					},
+					"configurationValues": {
+						TypeSpec:    schema.TypeSpec{Type: "string"},
+						Description: "Custom configuration values for addons with single JSON string. This JSON string value must match the JSON schema derived from describe-addon-configuration.",
```

Comment:
Why don’t we change this to a map and serialize it for users?

Reply:
Updated this to be a map and added additional integration tests for this.